test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetUITestCase.cs

```diff
@@ -149,6 +156,32 @@ public void UpdatePackageFromUI()
             CommonUtility.AssertPackageInPackagesConfig(VisualStudio, project, "newtonsoft.json", "10.0.3", XunitLogger);
         }
 
+        private void OpenNuGetPackageManagerWithDte()
+        {
+            VisualStudio.ObjectModel.Solution.WaitForOperationsInProgress(TimeSpan.FromMinutes(3));
+            WaitForCommandAvailable("Project.ManageNuGetPackages", TimeSpan.FromMinutes(1));
+            VisualStudio.Dte.ExecuteCommand("Project.ManageNuGetPackages");
+        }
+
+        private void WaitForCommandAvailable(string commandName, TimeSpan timeout)
+        {
+            WaitForCommandAvailable(VisualStudio.Dte.Commands.Item(commandName), timeout);
+        }
+
+        private void WaitForCommandAvailable(Command cmd, TimeSpan timeout)
+        {
+            var stopWatch = new Stopwatch();
+            stopWatch.Start();
+            while (stopWatch.Elapsed < timeout)
+            {
+                if (cmd.IsAvailable)
+                {
+                    return;
+                }
+                System.Threading.Thread.Sleep(250);
+            }
+        }
```

Comment:
How about logging a warning that command X wasn't available even after waiting Y seconds?